ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,219 @@
+          </div>
+        </nuxeo-dialog>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-picker';
+    }
+
+    static get properties() {
+      return {
+        /**
+         * The aggregations returned by `provider`.
+         */
+        aggregations: Object,
+        /**
+         * List of content enrichers passed on to `provider`.
+         */
+        enrichers: String,
+        /**
+         * The number of results per page.
+         */
+        pageSize: Number,
+        /**
+         * The query parameters passed on to `provider`.
+         */
+        params: Object,
+        /**
+         * The id of the `nuxeo-page-provider` instance used to perform the search.
+         */
+        provider: String,
+        /**
+         * Name of the quick filters to be displayed in case you don't want to display all of them.
+         * Expected format : ['quickfilter1','quickfilter2'].
+         */
+        quickFilters: Object,
+        /**
+         * List of comma separated values of the document schemas to be returned.
+         * All document schemas are returned by default.
+         */
+        schemas: String,
+        /**
+         * The name of the search layout.
+         */
+        searchName: String,
+      };
+    }
+
+    ready() {
+      super.ready();
+      if (!this._nxProvider) {
+        this._nxProvider = this.$.provider;
+      }
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      if (this.listenedResults) {
+        this.listenedResults.removeEventListener('selected-items-changed', this._updateInsertButton.bind(this));
+      }
+    }
+
+    open() {
+      if (this.$.resultsView) {
+        // XXX could make sense to consider having this as a public method on nuxeo-results-view
+        this.$.resultsView._clear();
+      }
+      this._updateInsertButton();
+      this.$.dialog.open();
+    }
+
+    close() {
+      this.$.dialog.close();
+    }
+
+    _getSelectedItems() {
+      return (
+        this.$.resultsView &&
+        this.$.resultsView.results &&
+        this.$.resultsView.results.results &&
+        this.$.resultsView.results.results.selectedItems
+      );
+    }
+
+    _updateInsertButton() {
+      const selectedItems = this._getSelectedItems();
+      this.$.insertButton.disabled = !(selectedItems && selectedItems.length);
+    }
+
+    _onInsert() {
+      const selectedItems = this._getSelectedItems();
+      if (selectedItems) {
+        this.dispatchEvent(
+          new CustomEvent('inserted', {
+            composed: true,
+            bubbles: true,
+            detail: { selectedItems },
+          }),
+        );
+        this.close();
+      }
+    }
+
+    _onNavigate(e) {
+      // TODO check with other types of listing like tables and justified grid
+      e.composedPath()[0]
+        .$$('paper-icon-button')
+        .click();
+    }
+
+    _onResultsChanged(e) {
+      if (e.detail.value) {
```

Comment:
[The `results` on `<nuxeo-search-results-layout>` is already set to `notify`](https://github.com/nuxeo/nuxeo-elements/blob/ELEMENTS-1311-move-search-layouts/ui/search/nuxeo-search-results-layout.js#L80).
It won't bubble through `<nuxeo-results-view>` because [property change events don't bubble](https://polymer-library.polymer-project.org/3.0/docs/devguide/properties#notify).

Reply:
yep, but if you add the property on `nuxeo-results-view` and bind it to the one on `nuxeo-results-view` then it'll be a property change on `nuxeo-results-view` itself allowing you to listen to the events.

Reply:
>if you add the property on `nuxeo-results-view` and bind it to the one on `nuxeo-results-view`

Do you mean "the one on `nuxeo-search-results-layout`"? That doesn't seem to work because `nuxeo-search-results-layout` also doesn't have a `selectedItems`. At this point the hierarchy looks like:
```
<nuxeo-document-picker>
  <nuxeo-results-view>
    <nuxeo-search-results-layout>      <!-- has direct access to <nuxeo-results> down below -->
      <nuxeo-layout>                   <!-- doesn't have selectedItems -->
        <nuxeo-picker-search-results>  <!-- has selectedItems -->
          <nuxeo-results>              <!-- has selectedItems -->
```
The only reason we're able to use something like `this.$.resultsView.results.results` in the picker element is because `nuxeo-search-results-layout` has a `results` object (`nuxeo-results` instance) that bypasses `nuxeo-layout` and `nuxeo-picker-search-results` entirely.
Unfortunately I don't think we can "bubble up the prop binding" easily. Please let me know if I'm missing something here.